services/worker/src/worker/job_runners/split/first_rows_from_parquet.py

```diff
@@ -192,6 +192,7 @@ def __init__(
             httpfs=HTTPFileSystem(headers={"authorization": f"Bearer {self.app_config.common.hf_token}"}),
             unsupported_features=[],
             all_columns_supported_datasets_allow_list="all",
+            max_arrow_data_in_memory=app_config.rows_index.max_arrow_data_in_memory,
```

Comment:
catch the TooBigRows exception and raise a CacheableError instead, so that we have an X-Error-Code header.